ultralytics/yolo/v8/detect/predict.py

```diff
@@ -14,7 +14,8 @@ def get_annotator(self, img):
         return Annotator(img, line_width=self.args.line_thickness, example=str(self.model.names))
 
     def preprocess(self, img):
-        img = torch.from_numpy(img).to(self.model.device)
+        if not isinstance(img, torch.Tensor):
+            img = torch.from_numpy(img).to(self.model.device)
```

Comment:
@AyushExel probably this should be updated to:
```python
img = (torch.from_numpy(img) if not isinstance(img, torch.Tensor) else img).to(self.model.device)
```
or the `img` is not on the correct device when it's not tensor type.

Reply:
Ahh yes. Updated